lib/bibliothecary/multi_parsers/dependencies_csv.rb

```diff
@@ -0,0 +1,151 @@
+require 'csv'
+
+module Bibliothecary
+  module MultiParsers
+    module DependenciesCSV
+      include Bibliothecary::Analyser
+      include Bibliothecary::Analyser::TryCache
+
+      def self.mapping
+        {
+          match_filename('dependencies.csv') => {
+            kind: 'lockfile',
+            parser: :parse_dependencies_csv
+          }
+        }
+      end
+
+      # Processing a CSV file isn't as exact as using a real manifest file,
+      # but you can get pretty close as long as the data you're importing
+      # is simple.
+      class CSVFile
+        # Header structures are:
+        #
+        # <field to fill in for dependency> => {
+        #   match: [<regexp of incoming column name to match in priority order>...],
+        #   [default]: <optional default value for this field>
+        # }
+        HEADERS = {
+          "platform" => {
+            match: [
+              /^platform$/i
+            ]
+          },
+          "name" => {
+            match: [
+              /^name$/i
+            ]
+          },
+          # The "requirement" column in this case is going to want exact
+          # versions. Prefer those over requirements, and don't use the
+          # Lockfile Requirement column from CSV exports.
+          "requirement" => {
+            match: [
+              /^(manifest |)requirement$/i,
+              /^version$/i,
+            ],
+            # strip anything from the front that doesn't look like a version number.
+            process: lambda do |value|
+              value.gsub(/^[^\dvV]*/, '')
```

Comment:
Versions we've seen in the wild include "HEAD", "Test", "Houston", "any sha hexdigest", "some-tag-name", "r7", "A", and so forth.
I think it's safe to remove "operator" characters (the list we've used elsewhere is `"~", "^", "<", ">", "*", "="` )

I feel like doing this may just not be necessary though, if we require everyone to keep the "contract" of the data...

manifest_requirement is defined to be an expression, right? so we can keep it as-is and use it as "requirement"
lockfile_requirement is defined to be a resolved version number without operators?

I'm not sure of the exact naming/details in this context but I do think we should try to define what each variable or column has in it, rather than heuristically parsing/stripping anything. Especially in this "raw" data interchange format that we can define in any way we like? If it's an expression then leave it, if it's a version then leave that, and basically we cannot reliably convert between the two, only the package manager can.

Maybe a goal is that if you import a package.json/package-lock.json, then download the CSV, then upload the CSV, you get the same answer - that seems to imply not messing with the text of any fields.